src/controllers/tileController.ts

```diff
@@ -1,28 +1,49 @@
-import { Request, Response } from "express";
+import { NextFunction, Request, Response } from "express";
 import { AppLocals } from "../types/app";
+import asyncControllerHandler from "../errors/asyncControllerHandler";
+import notFound from "../errors/notFound";
+import { GroutError } from "../errors/groutError";
+import { ErrorType } from "../errors/errorType";
+
+const parseIntParam = (param: string): number => {
+    // Native parseInt is not strict (ignores whitespace and trailing chars) so test with a regex
+    if (!/^\d+$/.test(param)) {
+        throw new GroutError(
+            `"${param}" is not an integer`,
+            ErrorType.BAD_REQUEST
+        );
+    }
+    return parseInt(param, 10);
```

Comment:
dont have to change this but you could use `Number(string)` which does everything you want without you having to write regex, `Number` evals the whole string vs `parseInt` stopping

Reply:
What would parseInt(Number(string)) do?

Reply:
you can do `Number.isInteger(Number(string))` for that, so something like
```js
const num = Number(string);
if (!Number.IsInteger) {
  throw...
}
return num
```
guess its about the same XD up to you

Reply:
OK, I might just leave it as it is. Nice idea though, thanks!